tests/ballerina-test-integration/src/test/java/org/ballerinalang/test/service/grpc/sample/UnaryBlockingHeaderTestCase.java

```diff
@@ -0,0 +1,82 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.test.service.grpc.sample;
+
+import org.ballerinalang.launcher.util.BCompileUtil;
+import org.ballerinalang.launcher.util.BRunUtil;
+import org.ballerinalang.launcher.util.CompileResult;
+import org.ballerinalang.model.values.BString;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.test.context.BallerinaTestException;
+import org.ballerinalang.test.context.ServerInstance;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * Test class for headers for gRPC unary service with blocking and non-blocking client.
+ *
+ */
+public class UnaryBlockingHeaderTestCase {
+
+    private ServerInstance ballerinaServer;
+
+    @BeforeClass
+    private void setup() throws Exception {
+        ballerinaServer = ServerInstance.initBallerinaServer(9090);
+        Path serviceBalPath = Paths.get("src", "test", "resources", "grpc", "unary-service3.bal");
+        ballerinaServer.startBallerinaServer(serviceBalPath.toAbsolutePath().toString());
+    }
+
+    @Test
+    public void testBlockingBallerinaClient() {
+
+        Path balFilePath = Paths.get("src", "test", "resources", "grpc", "unary-client.bal");
+        CompileResult result = BCompileUtil.compile(balFilePath.toAbsolutePath().toString());
+        BString request = new BString("WSO2");
+        final String serverMsg = "Hello WSO2";
+
+        BValue[] responses = BRunUtil.invoke(result, "testUnaryBlockingClient", new BValue[]{request});
+        Assert.assertEquals(responses.length, 1);
+        Assert.assertTrue(responses[0] instanceof BString);
+        Assert.assertEquals(responses[0].stringValue(), "Client got response: " + serverMsg);
+    }
+
+    @Test
+    public void testIntBlockingBallerinaClient() {
+
+        Path balFilePath = Paths.get("src", "test", "resources", "grpc", "unary-client.bal");
```

Comment:
All bal files, even in test cases and samples should follow naming conventions. All simples with words separated by underscore.